packages/lib-classifier/src/store/SubjectStore/shared/TextSubject.js

```diff
@@ -0,0 +1,53 @@
+import { flow, types } from 'mobx-state-tree'
+import asyncStates from '@zooniverse/async-states'
+
+const TextSubject = types
+  .model('TextSubject', {
+    content: types.maybeNull(types.string),
+    contentLoadingState: types.optional(types.enumeration('contentLoadingState', asyncStates.values), asyncStates.initialized),
+    error: types.optional(types.maybeNull(types.frozen({})), null)
+  })
+  .actions(self => {
+    function afterAttach () {
+      if (self.contentLoadingState === asyncStates.initialized) {
+        self.fetchContent()
+      }
+    }
+
+    function getSubjectUrl () {
+      // Find locations that have a text/plain MIME type.
+      const textLocation = self.locations.find(l => l['text/plain']) || {}
+      const url = Object.values(textLocation)[0]
+      if (url) {
+        return url
+      } else {
+        throw new Error('No text url found for this subject')
+      }
+    }
+
+    function * fetchContent () {
+      self.contentLoadingState = asyncStates.loading
+
+      try {
+        const url = getSubjectUrl()
+        const response = yield fetch(url)
+        if (!response.ok) {
+          const error = new Error(response.statusText)
+          error.status = response.status
+          throw error
+        }
+        self.content = yield response.text()
+        self.contentLoadingState = asyncStates.success
+      } catch (error) {
+        self.contentLoadingState = asyncStates.error
+        self.error = error
+      }
+    }
```

Comment:
Would it be worth moving this into a hook in the text viewer? Similar to the hook that fetches images for the image viewer, or the hook that fetches JSON for the data viewers.

Reply:
Yes! I have this on my short list of DigiLeap follow-up todos/clean-up. I haven't thought through how/if the hook would be used with the `textFromSubject` task and not duplicate location requests, I don't think there'll be any issue and it's possible (to request location once for viewer and task), just haven't thought through it utilizing a hook yet. I think requesting the text location with a hook similar to the other locations is what's best though. I'll create an issue so it doesn't get lost in my notes.